exec/cnex/cell.c

```diff
@@ -239,6 +247,71 @@ BOOL cell_arrayElementExists(const Cell *a, const Cell *e)
     return FALSE;
 }
 
+TString *cell_toString(Cell *c)
+{
+    assert(c->type != cNothing);
+    TString *r = string_newString();
+
+    switch (c->type) {
+        case cAddress:
+            break;
+        case cArray:
+        {
+            size_t x, y;
+            /* ToDo: Fix this to use proper string appends!! */
+            char *s = malloc(STRING_BUFFER_SIZE);
+            if (s == NULL) {
+                fatal_error("Could not allocate %d byte string buffer for array", STRING_BUFFER_SIZE);
+            }
+            strcpy(s, "[");
+            for (x = 0; x < c->array->size; x++) {
+                if (strlen(s) > 1) {
+                    strcat(s, ", ");
+                }
+                if (c->array->data[x].type == cString) {
+                    strcat(s, "\"");
+                    TString *v = cell_toString(&c->array->data[x]);
+                    for (y = 0; y < v->length; y++) {
+                        strcat(s, &v->data[y]);
+                    }
+                    strcat(s, "\"");
+                    string_freeString(v);
+                } else if (c->array->data[x].type == cNumber) {
+                    TString *v = cell_toString(&c->array->data[x]);
+                    char *sz = string_asCString(v);
+                    strcat(s, sz);
+                    free(sz);
+                    string_freeString(v);
+                }
+            }
+            strcat(s, "]");
+            r = string_appendCString(r, s);
+            free(s);
+            break;
+        }
+        case cBoolean:
+            r = string_appendCString(r, (c->boolean ? "TRUE" : "FALSE"));
+            break;
+        case cNothing:
```

Comment:
You have already asserted at the top of this function that the type is not cNothing.

Reply:
Removed this case.